src/standalone/Vegetation/spatially_varying_parameters.jl

```diff
@@ -305,3 +306,124 @@ function clm_medlyn_g1(
+    )
+
+Read spatially-varying canopy height data from CLM vegetation properties.
+
+Returns a Field of raw canopy heights from the CLM dataset without any modifications.
+For heights that respect atmospheric coupling constraints, use `effective_canopy_height`
+to cap the heights appropriately.
+
+# Arguments
+- `surface_space`: The surface space for regridding
+
+# Optional Arguments
+- `regridder_type`: Type of regridder (default: `:InterpolationsRegridder`)
+- `extrapolation_bc`: Boundary conditions for extrapolation
+- `interpolation_method`: Spatial interpolation method (default: Linear)
+- `lowres`: Use low-resolution CLM data
+
+# Returns
+- Field of canopy heights in meters
+
+# Example
+```julia
+raw_height = clm_canopy_height(surface_space)
+effective_height = effective_canopy_height(raw_height, 10.0; buffer=2.0)
+```
+"""
+function clm_canopy_height(
+    surface_space;
+    regridder_type = :InterpolationsRegridder,
+    extrapolation_bc = (
+        Interpolations.Periodic(),
+        Interpolations.Flat(),
+        Interpolations.Flat(),
+    ),
+    interpolation_method = Interpolations.Linear(),
+    lowres = ClimaLand.Domains.use_lowres_clm(surface_space),
+)
+    context = ClimaComms.context(surface_space)
+    clm_artifact_path = Artifacts.clm_data_folder_path(; context, lowres)
+
+    canopy_height = SpaceVaryingInput(
+        joinpath(clm_artifact_path, "vegetation_properties_map.nc"),
+        "z_top",
+        surface_space;
+        regridder_type,
+        regridder_kwargs = (; extrapolation_bc, interpolation_method),
+    )
+    return canopy_height
+end
+
+"""
+    effective_canopy_height(
+        canopy_height::ClimaCore.Fields.Field,
+        z_atm::FT;
+        buffer::FT = FT(2.0)
+    ) where {FT}
+
+Caps canopy heights to ensure they remain below the atmospheric reference height.
+This is necessary when atmospheric forcing data (e.g., ERA5) is provided at a fixed
+reference height (typically 10m). The canopy height must be less than z_atm to ensure
+proper flux calculations and atmospheric coupling.
+
+# Arguments
+- `canopy_height`: Field of canopy heights read from CLM data (meters)
+- `z_atm`: Atmospheric reference height for forcing data (meters)
+- `buffer`: Minimum distance between canopy top and atmospheric reference (default: 2m)
+
+# Returns
+- Field of effective canopy heights, capped at `z_atm - buffer`
+
+# Example
+```julia
+raw_height = clm_canopy_height(surface_space)
+z_atm = FT(10.0)  # ERA5 forcing at 10m
+height = effective_canopy_height(raw_height, z_atm)
+```
+
+# Notes
+The function will log a warning message indicating how many grid cells had their
+canopy height capped and provide statistics about the original height distribution.
+This typically affects ~7% of global cells, primarily in regions with tall forests
+(e.g., tropical rainforests, temperate forests in Patagonia and New Zealand).
+"""
+function effective_canopy_height(
```

Comment:
I think this OK for now! In the future, we can adjust it. because, what actually matters (at least in standard Monin-Obukhov, I think) is not quite z_atm - h_canopy, but z_atmos - displacement height. so I think this will transform to be 
```
d = displacement_height(canopy_height..., LAI, maxLAI,....)
effective_d = min(d, z_atm - buffer)
```
and perhaps we even compute this when we compute the surface fluxes, not as an a-priori step (esp if displacement height changes in time)